zerver/webhooks/gitlab/view.py

```diff
@@ -359,10 +359,9 @@ def get_event(request, payload, branches):
         action = payload.get('object_attributes').get('action')
         event = "{} {}".format(event, action)
     elif event == 'Push Hook':
-        if branches is not None:
-            branch = get_branch_name(payload)
-            if branch not in branches.split(','):
-                return None
+        branch = get_branch_name(payload)
+        if branches is not None and branches.find(branch) == -1:
```

Comment:
I think we should leave the `if branches is not None` outside the call to `get_branch_name`, since there's no reason to compute it if we don't use it.